hummingbot/connector/exchange/chainflip_lp/chainflip_lp_api_order_book_data_source.py

```diff
@@ -0,0 +1,90 @@
+from hummingbot.core.data_type.order_book_message import OrderBookMessage
+from hummingbot.core.data_type.order_book_tracker_data_source import OrderBookTrackerDataSource
+from hummingbot.core.event.event_forwarder import EventForwarder
+from hummingbot.core.web_assistant.connections.data_types import WSJSONRequest
+from hummingbot.core.event.events import OrderBookEvent
+from hummingbot.core.web_assistant.web_assistants_factory import WebAssistantsFactory
+from hummingbot.core.web_assistant.ws_assistant import WSAssistant
+
+if TYPE_CHECKING:
+    from hummingbot.connector.exchange.chainflip_lp.chainflip_lp_exchange import ChainflipLPExchange
+
+
+class ChainflipLPAPIOrderBookDataSource(OrderBookTrackerDataSource):
+    def __init__(
+        self,
+        trading_pairs: List[str],
+        connector: "ChainflipLPExchange",
+        data_source: "ChainflipLPDataSource",
+        api_factory: "WebAssistantsFactory",
+        domain: str = CONSTANTS.DEFAULT_DOMAIN,
+
+    ):
+        super().__init__(trading_pairs=trading_pairs)
+        self._ev_loop = asyncio.get_event_loop()
+        self._connector = connector
+        self._data_source = data_source
+        self._domain = domain
+        self._api_factory = api_factory
+        # self._forwarders = []
+        # self._configure_event_forwarders()
+
+    async def get_last_traded_prices(self,
+                                     trading_pairs: List[str],
+                                     domain: Optional[str] = None) -> Dict[str, float]:
+        return await self._connector.get_last_traded_prices(trading_pairs=trading_pairs)
+    async def _order_book_snapshot(self, trading_pair: str) -> OrderBookMessage:
+        snapshot = await self._data_source.order_book_snapshot(trading_pair=trading_pair)
+        return snapshot
+    async def listen_for_subscriptions(self):
+        pass
+
+    async def _parse_order_book_snapshot_message(self, raw_message: Dict[str, Any], message_queue: asyncio.Queue):
+        raise NotImplementedError
+
+    async def _connected_websocket_assistant(self) -> WSAssistant:
+        ws: WSAssistant = await self._api_factory.get_ws_assistant()
+        await ws.connect(ws_url=CONSTANTS.WS_RPC_URLS[self._domain],
+                         ping_timeout=CONSTANTS.WS_HEARTBEAT_TIME_INTERVAL)
+        return ws
+
+    async def _subscribe_channels(self, ws: WSAssistant):
+        """
+        Subscribes to the scheduled swaps
+        """
+        try:
+            all_assets = self._data_source.assets_list()
+            for trading_pair in self._trading_pairs:
+                symbol = await self._connector.exchange_symbol_associated_to_pair(trading_pair=trading_pair)
+                asset = DataFormatter.format_trading_pair(symbol, all_assets)
+                payload = {
+                    "id":1,
+                    "jsonrpc":"2.0",
+                    "method": CONSTANTS.SCHEDULED_SWAPS,
```

Comment:
this will subscribe to the swaps that are scheduled to be executed in 2 blocks (12 seconds). im not sure if that breaks any assumptions, but i wanted to add the context

Reply:
I am still trying to figure out  the use case that might go with this subscription so it might removed soon in case there is none